backend/src/controllers/MyRestaurantController.ts

```diff
@@ -45,7 +46,47 @@ const getMyRestaurant = async (req: Request, res: Response, next: NextFunction)
   }
 };
 
+const getMyRestaurantOrders = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const restaurant = await Restaurant.findOne({ user: req.userId });
+    if (!restaurant) {
+      throw new AppError(`Restaurant not found for user ID: ${req.userId}`, 404);
+    }
+    const orders = await Order.find({ restaurant: restaurant._id })
+      .populate('restaurant')
+      .populate('user');
+    res.status(200).json(orders);
+  } catch (error) {
+    next(error); // Pass error to the centralized handler
+  }
+};
+
+const updateOrderStatus = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { orderId } = req.params;
+    const { status } = req.body;
+
+    const order = await Order.findById(orderId);
+    if (!order) {
+      throw new AppError(`Order not found for ID: ${orderId}`, 404);
+    }
+
+    const restaurant = await Restaurant.findById(order.restaurant);
+    if (restaurant?.user.toString() !== req.userId) {
+      throw new AppError('Unauthorized', 401);
+    }
+
+    order.status = status;
+    await order.save();
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Add error handling for order.save() operation</b></div>
<div id="fix">

Consider adding error handling for the `order.save()` operation. If the save fails, the current implementation might not catch the error.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -78,8 +78,13 @@ const updateOrderStatus = async (req: Request, res: Response, next: NextFunction
 
      order.status = status;
 -    await order.save();
 -    res.status(200).json(order);
 +    try {
 +      await order.save();
 +      res.status(200).json(order);
 +    } catch (saveError) {
 +      console.error('Error saving order:', saveError);
 +      throw new AppError('Failed to update order status', 500);
 +    }
    } catch (error) {
      next(error);
    }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/49#issuecomment-2440142983>#53f2e3</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged